src/NuGetSdkResolver.UnitTests/NuGetSdkResolver_Tests.cs

```diff
@@ -101,6 +104,38 @@ public void TryGetNuGetVersionNoVersionSpecified()
                 context: context);
         }
 
+#if FEATURE_APPDOMAIN
+        /// <summary>
+        /// Verifies that when an SDK version is null that no NuGet assemblies are loaded.  This helps ensure that we're not loading
+        /// extra assemblies unless they are needed.  A lot of private classes exist in the NuGetSdkResolver in order to make sure
+        /// that types are loaded until they are needed.
+        /// </summary>
+        [Fact]
+        private void TryGetNuGetVersionNullVersionShouldNotLoadNuGetAssemblies()
```

Comment:
Is it worth writing a test that uses reflection to peek at the AssemblyResolve handlers and assert their number does not change after a nuget resolver call with a non-null version? :)